firebase-common/src/test/java/com/google/firebase/heartbeatinfo/DefaultHeartBeatInfoTest.java

```diff
@@ -15,20 +15,50 @@
 package com.google.firebase.heartbeatinfo;
 
 import static com.google.common.truth.Truth.assertThat;
+import static org.mockito.ArgumentMatchers.anyBoolean;
 import static org.mockito.ArgumentMatchers.anyLong;
 import static org.mockito.ArgumentMatchers.anyString;
 import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.when;
 
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import org.junit.Before;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.junit.runners.JUnit4;
 
 @RunWith(JUnit4.class)
 public class DefaultHeartBeatInfoTest {
   private String testSdk = "fire-test";
+  private ExecutorService executor;
+  private TestOnCompleteListener<Boolean> storeOnCompleteListener;
+  private TestOnCompleteListener<List<HeartBeatResult>> getOnCompleteListener;
   private HeartBeatInfoStorage storage = mock(HeartBeatInfoStorage.class);
-  private DefaultHeartBeatInfo heartBeatInfo = new DefaultHeartBeatInfo(storage);
+  private Set<HeartBeatConsumer> logSources =
+      new HashSet<HeartBeatConsumer>() {
+        {
+          add(new HeartBeatConsumer() {});
+        }
+      };
+  private DefaultHeartBeatInfo heartBeatInfo;
+
+  @Before
+  public void setUp() {
+    executor = new ThreadPoolExecutor(0, 1, 30L, TimeUnit.SECONDS, new LinkedBlockingQueue<>());
+    storeOnCompleteListener = new TestOnCompleteListener<>();
```

Comment:
what is the purpose of these listeners? why not just use `Tasks.await(info.storeHeartBeatInfo(testSdk))` for example?

Reply:
hmm I used Tasks.await before but it would cause the looper wait issue as there could be more than one threads. Hence the need of the listener similar to what has been done in firebase-installations.